site/content/extend/plugins/webapp/actions.md

```diff
@@ -1,115 +1,493 @@
+                file_ids: fileIds,
+            };
+
+            actions.push({
+                type: FileTypes.RECEIVED_FILES_FOR_POST,
+                postId: pendingPostId,
+                data: files,
+            });
+        }
+
+        actions.push({
+            type: PostTypes.RECEIVED_NEW_POST,
+            data: {
+                ...newPost,
+                id: pendingPostId,
+            },
+        });
 
-    render() {
-        return (
-            <div>
-                {'First name: ' + this.props.user.first_name}
-                <a
-                    href='#'
-                    onClick={this.updateFirstName}
-                >
-                    Click me to update the first name!
-                </a>
-            </div>
-        );
-    }
+```
+
+**getMyTeams**
+
+Dispatch this action to fetch all the team types associated with the current user.
+
+```javascript
+export function getMyTeams(): ActionFunc {
+    return bindClientFunc({
+        clientFunc: Client4.getMyTeams,
+        onRequest: TeamTypes.MY_TEAMS_REQUEST,
+        onSuccess: [TeamTypes.RECEIVED_TEAMS_LIST, TeamTypes.MY_TEAMS_SUCCESS],
+        onFailure: TeamTypes.MY_TEAMS_FAILURE,
+    });
+}
+```
+
+**createUser**
+
+Dispatch this action to create a new user profile.
+
+```javascript
+export function createUser(user: UserProfile, token: string, inviteId: string, redirect: string): ActionFunc {
+    return async (dispatch: DispatchFunc, getState: GetStateFunc) => {
+        let created;
+
+        try {
+            created = await Client4.createUser(user, token, inviteId, redirect);
+        } catch (error) {
+            forceLogoutIfNecessary(error, dispatch, getState);
+            dispatch(logError(error));
+            return {error};
+        }
+
+        const profiles: {
+            [userId: string]: UserProfile;
+        } = {
+            [created.id]: created,
+        };
+        dispatch({type: UserTypes.RECEIVED_PROFILES, data: profiles});
+
+        return {data: created};
+    };
 }
 ```
 
-This component will display a user's first name and then, when the link is clicked, use an action to update that user's first name to "Jim".
+## Some common selectors
 
-The action `patchUser` is from mattermost-redux. It takes in a subset of a user object and updates the user on the server, [using the `PUT /users/{user_id}/patch` endpoint](https://api.mattermost.com/#tag/users%2Fpaths%2F~1users~1%7Buser_id%7D~1patch%2Fput).
+We've listed out some of the commonly-used selectors that you can use in your web app plugin. You can find all the selectors that are available for your plugin to import [in the source code for mattermost-redux](https://github.com/mattermost/mattermost-redux/tree/master/src/selectors).
 
-We must now use our container to import this action and pass it our component. Open up the `index.js` file and add:
+**getCurrentUserId**
+
+Retrieves the `userId` of the current user from the `Mattermost Server`.
```

Comment:
Selectors retrieve from the Redux store. Client actions return from the Mattermost server.

Reply:
Thanks for pointing this out. Making the changes now.